doc/source/data/api/input_output.rst

```diff
@@ -1,150 +1,149 @@
+
 .. autofunction:: ray.data.range_table
+
 .. autofunction:: ray.data.range_tensor
+
+Python Objects
+--------------
+
+.. autofunction:: ray.data.from_items
+
+CSV
+---
+
 .. autofunction:: ray.data.read_csv
+
+.. automethod:: ray.data.Dataset.write_csv
+    :noindex:
+
+JSON
+----
+
 .. autofunction:: ray.data.read_json
-.. autofunction:: ray.data.read_parquet
-.. autofunction:: ray.data.read_parquet_bulk
-.. autofunction:: ray.data.read_numpy
-.. autofunction:: ray.data.read_text
-.. autofunction:: ray.data.read_binary_files
-.. autofunction:: ray.data.read_datasource
-.. autofunction:: ray.data.from_items
-.. autofunction:: ray.data.from_arrow
-.. autofunction:: ray.data.from_arrow_refs
-.. autofunction:: ray.data.from_huggingface
-.. autofunction:: ray.data.from_spark
-.. autofunction:: ray.data.from_dask
-.. autofunction:: ray.data.from_modin
-.. autofunction:: ray.data.from_mars
+
+.. automethod:: ray.data.Dataset.write_json
+    :noindex:
+
+Pandas
+------
+
 .. autofunction:: ray.data.from_pandas
+
 .. autofunction:: ray.data.from_pandas_refs
-.. autofunction:: ray.data.from_numpy
-.. autofunction:: ray.data.from_numpy_refs
 
-.. _dataset-api:
+.. automethod:: ray.data.Dataset.to_pandas
+    :noindex:
 
-Dataset API
------------
+.. automethod:: ray.data.Dataset.to_pandas_refs
+    :noindex:
 
-.. autoclass:: ray.data.Dataset
-    :members:
+NumPy
+-----
 
-.. _dataset-pipeline-api:
+.. autofunction:: ray.data.read_numpy
 
-Block API
----------
+.. autofunction:: ray.data.from_numpy
 
-.. autoclass:: ray.data.block.BlockExecStats
-    :members:
+.. autofunction:: ray.data.from_numpy_refs
 
-.. autoclass:: ray.data.block.BlockMetadata
-    :members:
+.. automethod:: ray.data.Dataset.write_numpy
+    :noindex:
 
-.. autoclass:: ray.data.block.BlockAccessor
-    :members:
+.. automethod:: ray.data.Dataset.to_numpy_refs
+    :noindex:
 
-DatasetContext API
-------------------
+Parquet
+-------
 
-.. autoclass:: ray.data.context.DatasetContext
-    :members:
+.. autofunction:: ray.data.read_parquet
 
-DatasetPipeline API
--------------------
+.. autofunction:: ray.data.read_parquet_bulk
 
-.. autoclass:: ray.data.dataset_pipeline.DatasetPipeline
-    :members:
+.. automethod:: ray.data.Dataset.write_parquet
+    :noindex:
 
-GroupedDataset API
-------------------
+Arrow
+-----
 
-.. autoclass:: ray.data.grouped_dataset.GroupedDataset
-    :members:
+.. autofunction:: ray.data.from_arrow
 
-Aggregate API
--------------
+.. autofunction:: ray.data.from_arrow_refs
 
-.. autoclass:: ray.data.aggregate.AggregateFn
-    :members:
+.. automethod:: ray.data.Dataset.to_numpy_refs
+    :noindex:
 
-.. autoclass:: ray.data.aggregate.Count
-    :members:
+Text
+----
 
-.. autoclass:: ray.data.aggregate.Sum
-    :members:
+.. autofunction:: ray.data.read_text
 
-.. autoclass:: ray.data.aggregate.Max
-    :members:
+Binary
+------
 
-.. autoclass:: ray.data.aggregate.Mean
-    :members:
+.. autofunction:: ray.data.read_binary_files
```

Comment:
If we're ordering by popularity, I think that the text and binary sections should come before the Arrow section.

Another option is to order/group by file I/O vs. to/from in-memory data, although NumPy is both, so this delineation isn't clean.

Reply:
> Another option is to order/group by file I/O vs. to/from in-memory data

Yeah I also agree on this. Maybe something like

```
Synthetic Data
Python Objects

Parquet
CSV
JSON
Text
Binary

Pandas
NumPy
Arrow

Dask
Spark
Modin
Mars
HuggingFace

Datasource API
Partitioning API
MetadataProvider API
```